test/manifest.ts

```diff
@@ -14,22 +14,564 @@
 
 import {describe, it, afterEach} from 'mocha';
 
-import {Manifest} from '../src/manifest';
+import {Manifest, ManifestConfig} from '../src/manifest';
 import * as sinon from 'sinon';
-import {GitHub} from '../src/github';
-import {stubFilesFromFixtures} from './releasers/utils';
+import {GitHub, MergedGitHubPRWithFiles} from '../src/github';
+import {stubFilesFromFixtures, buildGitHubFileRaw} from './releasers/utils';
 import {expect} from 'chai';
+import {Commit} from '../src/graphql-to-commits';
 import {CheckpointType} from '../src/util/checkpoint';
+import {stubSuggesterWithSnapshot, buildMockCommit} from './helpers';
 
 const fixturePath = './test/fixtures/manifest/repo';
 const defaultBranch = 'main';
+const lastReleaseSha = 'abc123';
 const sandbox = sinon.createSandbox();
 
 describe('Manifest', () => {
   afterEach(() => {
     sandbox.restore();
   });
 
+  function stubGithub(options: {
+    github: GitHub;
+    commits?: Commit[];
+    manifest?: string;
+    lastReleaseSha?: string;
+    mergedPRFiles?: string[];
+    addLabel?: string | false;
+  }) {
+    const {
+      github,
+      commits,
+      manifest,
+      lastReleaseSha,
+      mergedPRFiles,
+      addLabel,
+    } = options;
+    if (manifest !== undefined && lastReleaseSha) {
+      sandbox
+        .stub(github, 'getFileContentsWithSimpleAPI')
+        .withArgs('.release-please-manifest.json', lastReleaseSha, false)
+        .resolves(buildGitHubFileRaw(manifest));
+    }
+    let mergedPR: MergedGitHubPRWithFiles | undefined;
+    if (lastReleaseSha) {
+      mergedPR = {
+        sha: lastReleaseSha,
+        title: '',
+        body: '',
+        number: 22,
+        baseRefName: '',
+        headRefName: '',
+        files: mergedPRFiles ?? [],
+        labels: [],
+      };
+    }
+    sandbox.stub(github, 'lastMergedPRByHeadBranch').resolves(mergedPR);
+    sandbox
+      .stub(github, 'findFilesByFilename')
+      // implementation strips leading `path`
+      .resolves(['src/foolib/version.py']);
+    // creating a new PR
+    sandbox.stub(github, 'findOpenReleasePRs').resolves([]);
+    sandbox
+      .stub(github, 'commitsSinceSha')
+      .withArgs(lastReleaseSha)
+      .resolves(commits ?? []);
+
+    const addLabelsStub = sandbox.stub(github, 'addLabels');
+    const labelToAdd = addLabel ?? 'autorelease: pending';
+    if (labelToAdd) {
+      addLabelsStub.withArgs([labelToAdd], 22).resolves(true);
+    }
+    addLabelsStub.throws(() => {
+      const argsArr = addLabelsStub.getCalls().map(c => c.args);
+      const unexpected = argsArr.filter(args => args[0][0] !== labelToAdd);
+      return new Error(
+        'addLabelsStub called with unexpected args: ' +
+          JSON.stringify(unexpected[0])
+      );
+    });
+  }
```

Comment:
@AprilArcus here's what I landed on for the sinon.stub functionality that gives me an error when called with unexpected arguments and tells me what they were. I should probably be using a [mock](https://sinonjs.org/releases/latest/mocks/) but reading the philosophy at the top of that dock made me shy away from it. Specifically

> In general you should have no more than one mock (possibly with several expectations) in a single test.

because my follow on PR has a few more stub/mock thingy's that I want to tell me when they've been used unexpectedly

Reply:
haha, and now that I finally wrote that down I think I understand. the "mock" unit here would actually be the `github` instance and the "several expectations" would be expectations I set on methods and how they should be called. I'm gonna give that a spin.

@bcoe If you and @chingor13 are ok with this PR as is, I'll change this "stub abusage" into "mock usage" in the next PR